### PR TITLE
Rename two_qubit_matrix related functions

### PR DESCRIPTION
**Pull request by tonybruguier**
This could be the first step of solving issue #5041

### CODE REVIEW

File: cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py [hunk 1]
```diff
@@ -31,13 +32,25 @@
     import cirq
 
 
+@_compat.deprecated(fix='Please use two_qubit_matrix_to_cz_operations', deadline='v0.15')
```
Comment (tanujkhattar) [inline cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:35]:
```suggestion
@_compat.deprecated(fix='Please use cirq.two_qubit_matrix_to_cz_operations', deadline='v0.15')
```

File: cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py [hunk 2]
```diff
@@ -61,13 +74,29 @@ def two_qubit_matrix_to_operations(
     return operations
 
 
+@_compat.deprecated(
+    fix='Please use two_qubit_matrix_to_diagonal_and_cz_operations', deadline='v0.15'
```
Comment (tanujkhattar) [inline cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:78]:
```suggestion
    fix='Please use cirq.two_qubit_matrix_to_diagonal_and_cz_operations', deadline='v0.15'
```

File: cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py [hunk 3]
```diff
@@ -31,13 +32,25 @@
...
+@_compat.deprecated(fix='Please use two_qubit_matrix_to_cz_operations', deadline='v0.15')
 def two_qubit_matrix_to_operations(
     q0: 'cirq.Qid',
     q1: 'cirq.Qid',
     mat: np.ndarray,
     allow_partial_czs: bool,
     atol: float = 1e-8,
     clean_operations: bool = True,
+) -> List[ops.Operation]:
+    return two_qubit_matrix_to_cz_operations(q0, q1, mat, allow_partial_czs, atol, clean_operations)
```
Comment (tanujkhattar) [inline cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:44]: nit: Don't remove the docstring on the original deprecated method, here and below.

**Review by tanujkhattar — APPROVED** [review]
LGTM % nits. 

Feel free to submit after fixing the nits.

File: cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py (hunk 3 above)
Comment (tonybruguier) [reply cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:59]: Done.

**Comment by tonybruguier** [comment]
Thanks for the quick turn-around on the review, @tanujkhattar 

I addressed your comments, I think, and it looks like it could be good to go, but I don't have write permissions, so I think you would have to merge it (if you think it's OK to do so, of course).

**Pull request merged**